src/objects/Mesh.js

```diff
@@ -147,17 +147,48 @@ Mesh.prototype = Object.assign( Object.create( Object3D.prototype ), {
 
 		}
 
-		function checkIntersection( object, material, raycaster, ray, pA, pB, pC, point ) {
+		function checkIntersection( object, material, raycaster, ray, pA, pB, pC, point, faceIndex ) {
 
 			var intersect;
+			var faceMaterial;
+			
+			if ( material.length && faceIndex ) {
+				
+				var groups = object.geometry.groups;
+				
+				if ( groups ) {
+					
+					for ( var i = 0; i < groups.length; i++ ) {
+						
+						var group = groups[ i ];
+						
+						if ( faceIndex >= group.start && faceIndex <= group.start + group.count ) {
+							
+							faceMaterial = object.materials[ i ];
```

Comment:
Is there a `materials` property?

Is the material index of the `i`th group equal to `i`?